shelly/src/main/java/au/com/jtribe/shelly/Call.java

```diff
@@ -0,0 +1,75 @@
+package au.com.jtribe.shelly;
+
+import android.Manifest;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.RequiresPermission;
+
+/**
+ * Represents a phone call to a number using Intent.ACTION_CALL
+ */
+public class Call {
+
+  private String phoneNumber;
+
+  public Call() {
+  }
+
+  /**
+   * Adds a phone number to the Call object that will be called using the intent
+   *
+   * @param number Phone number that will be called
+   * @return Call object this method was called on for method chaining
+   */
+  @RequiresPermission(Manifest.permission.CALL_PHONE)
+  public Call number(String number) {
```

Comment:
Annotate the parameter with `@NonNull` and the method with `@CheckResult` and `@NonNull`